test/contracts/staking/staking.test.ts

```diff
@@ -0,0 +1,662 @@
+const EPOCH_DURATION = 60
+const START_DELAY = 15
+const REWARDS_AVAILABLE_OFFSET = 20
+const MIN_POOL_STAKE = 500
+const REWARD_TOKEN_1_AMOUNT = 4000
+
+describe('Staking Pool Tests', () => {
+    describe('Initialization', () => {
+        const ratio = 0
+        const amount = BigNumber.from(9090)
+        before(async () => {
+            rewardToken1 = await deployContract<ERC20PresetMinterPauser>(
+                'ERC20PresetMinterPauser',
+                'Another erc20 Token',
+                'SYM'
+            )
+        })
+        it('launch paused then unpause', async () => {
+            admin = (await signer(0)).address
+            user = await signer(1)
+            user2 = await signer(2)
+
+            const symbol = 'EEK'
+            stakeTokens = await deployContract<ERC20PresetMinterPauser>(
+                'ERC20PresetMinterPauser',
+                'Another erc20 Token',
+                symbol
+            )
+
+            const epochStartTimestamp = (await getTimestampNow()) + START_DELAY
+            const rewardsAvailableTimestamp =
+                REWARDS_AVAILABLE_OFFSET + epochStartTimestamp + EPOCH_DURATION
+            const stakingPoolInfo = {
+                daoId: 0,
+                minTotalPoolStake: MIN_POOL_STAKE,
+                maxTotalPoolStake: 600,
+                minimumContribution: 5,
+                epochDuration: EPOCH_DURATION,
+                epochStartTimestamp,
+                emergencyMode: false,
+                treasury: admin,
+                stakeToken: stakeTokens.address,
+                rewardType: RewardType.FLOATING,
+                rewardTokens: []
+            }
+
+            stakingPool = await deployContract('StakingPool')
+            await stakingPool.initialize(
+                stakingPoolInfo,
+                true,
+                rewardsAvailableTimestamp
+            )
+
+            await expect(
+                userDeposit(user, BigNumber.from(20))
+            ).to.be.revertedWith('Pausable: paused')
+
+            await increaseTime(START_DELAY)
+            await stakingPool.unpause()
+
+            expect(await userDeposit(user, BigNumber.from(5)))
+        })
+        it('initialize rewards fails no allowance', async () => {
+            await expect(
+                stakingPool.initializeRewardTokens(admin, [
+                    {
+                        tokens: rewardToken1.address,
+                        maxAmount: amount,
+                        ratio: 0
+                    }
+                ])
+            ).to.be.revertedWith('StakingPool: invalid allowance')
+        })
+        it('initialize rewards', async () => {
+            await rewardToken1.mint(admin, amount)
+            verifyInitializeRewardsEvent(
+                {rewardTokens: rewardToken1.address, amount},
+                await initializeRewards(rewardToken1, admin, amount, ratio)
+            )
+        })
+    })
+
+    describe('Floating Staking Tests', () => {
```

Comment:
nit; naming`floating rewards`

You should never need to have the term `tests` in descriptions ...in a test file 😁